### PR TITLE
[e2e] update README

### PR DESCRIPTION
**Pull request by collinjackson**
Updates README with some documentation updates.

### CODE REVIEW

File: packages/e2e/README.md [hunk 1]
```diff
@@ -27,12 +27,19 @@ void main() {
...
+`example/test/`. It is also acceptable to put e2e tests in `test_driver/` folder
+so that they're alongside the runner app (see below).
+
 ## Using Flutter driver to run tests
 
 `E2EWidgetsTestBinding` supports launching the on-device tests with `flutter drive`.
 Note that the tests don't use the `FlutterDriver` API, they use `testWidgets` instead.
 
-Put the a file named `<package_name>_test.dart` in the app' `test_driver` directory:
+Put the a file named `<package_name>_e2e_test.dart` in the app' `test_driver` directory:
```
Comment (digiter) [inline packages/e2e/README.md]: Does `flutter test` scans `_test.dart` in a directory and execute them?

File: packages/e2e/README.md [hunk 2]
```diff
@@ -106,22 +113,27 @@ dependencies {
 To e2e test on a local Android device (emulated or physical):
 
 ```
-./gradlew connectedAndroidTest -Ptarget=`pwd`/../test_driver/<package_name>_e2e.dart
+./gradlew app:connectedAndroidTest -Ptarget=`pwd`/../test_driver/<package_name>_e2e.dart
```
Comment (digiter) [inline packages/e2e/README.md]: The suffix should be `_e2e.dart` or `_e2e_test.dart` as above?

File: packages/e2e/README.md (hunk 1 above)
Comment (collinjackson) [reply packages/e2e/README.md]: Yes. It only looks in the test/ directory by default.

File: packages/e2e/README.md (hunk 2 above)
Comment (collinjackson) [reply packages/e2e/README.md]: The suffix here should be _e2e.dart. The _e2e_test.dart suffix is reserved for the driver runner (which I hope will be unnecessary in the future).

File: packages/e2e/README.md (hunk 1 above)
Comment (digiter) [reply packages/e2e/README.md]: Then when executing `flutter test` without an attached device, the `_e2e_test.dart` will execute. Is it expected?

Comment (collinjackson) [reply packages/e2e/README.md]: `flutter test` only looks for tests in the test/ folder. test_driver/ is ignored.

**Comment by collinjackson** [comment]
Merging this since you LGTM'd it but if you'd like to make more edits let me know.

**Pull request merged**